docs/orchestrating-elastic-stack-applications/maps.asciidoc

```diff
@@ -0,0 +1,245 @@
+      - name: maps
+        volumeMounts:
+        - name: map-data
+          readOnly: true
+          mountPath: /usr/src/app/data
+      volumes:
+        - name: map-data
+          persistentVolumeClaim:
+            claimName: ems-basemap
+
+----
+
+The name of the container in the Pod template must be `maps`.
+
+[id="{p}-maps-advanced-configuration"]
+== Advanced configuration
+
+If you already looked at the <<{p}-elasticsearch-specification,Elasticsearch on ECK>> documentation, some of these concepts might sound familiar to you.
+The resource definitions in ECK share the same philosophy when you want to:
+
+* Customize the Pod configuration
+* Customize the product configuration
+* Manage HTTP settings
+
+
+
+[id="{p}-maps-configuration"]
+=== {ems} configuration
+You can add any valid {ems} setting as documented on the link:https://www.elastic.co/guide/en/kibana/current/maps-connect-to-ems.html#elastic-maps-server-configuration[product] page to the `spec.config` section.
+
+The following example demonstrates how to set the log level to `debug`:
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: maps-sample
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  config:
+     logging.level: debug
+----
+
+[id="{p}-maps-scaling"]
+=== Scale out a {ems} deployment
+
+To deploy more than one instance of maps, all the instances must mount the data volume containing the basemap read only. Once that is the case scaling out is just a matter of increasing the `count` attribute.
+
+[id="{p}-maps-http-configuration"]
+== HTTP configuration
+
+[id="{p}-maps-http-publish"]
+=== Load balancer settings and TLS SANs
+
+By default a `ClusterIP` link:https://kubernetes.io/docs/concepts/services-networking/service/[service] is created and associated with the maps deployment.
+If you want to expose maps externally with a link:https://kubernetes.io/docs/concepts/services-networking/service/#loadbalancer[load balancer], it is recommended to include a custom DNS name or IP in the self-generated certificate.
+
+[source,yaml,subs="attributes"]
+----
+apiVersion: maps.k8s.elastic.co/v1alpha1
+kind: ElasticMapsServer
+metadata:
+  name: maps-sample
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRef:
+    name: "elasticsearch-sample"
+  http:
+    service:
+      spec:
+        type: LoadBalancer # default is ClusterIP
+    tls:
+      selfSignedCertificate:
+        subjectAltNames:
+        - ip: 1.2.3.4
+        - dns: maps.example.com
+----
+
+[id="{p}-maps-http-custom-tls"]
+=== Provide your own certificate
+
+If you want to use your own certificate, the required configuration is identical to Elasticsearch. See <<{p}-custom-http-certificate>>.
+
+[id="{p}-maps-http-disable-tls"]
+=== Disable TLS
+
+You can disable the generation of the self-signed certificate and hence disable TLS.
+
+[source,yaml,subs="attributes"]
```

Comment:
Same here, should we just add a link to https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-tls-certificates.html#k8s-disable-tls ?